File: Problem_2/problem2.js
let sum = 0

function fibonacci(n) {
  if (n <= 2) return n;
  return fibonacci(n-2) + fibonacci(n-1);
}

for (let i = 1; fibonacci(i) < 4000000; i++) {
  if (fibonacci(i)%2 == 0) {
    sum += fibonacci(i);
  }
}

console.log(sum)

File: 01-09/Problem_2/problem2.ex
defmodule Fib do
  def fibonacci_sequence do
    Stream.unfold({1,1}, fn {a,b} -> {a, {b, a + b}} end)
  end
end

Fib.fibonacci_sequence
  |> Stream.take_while(&(&1 < 4000000))
  |> Stream.filter(&(rem(&1, 2) ==0))
  |> Enum.sum
  |> IO.puts